src/backend/distributed/deparser/deparse_comment_stmts.c

```diff
@@ -0,0 +1,78 @@
+/*-------------------------------------------------------------------------
+ *
+ * deparse_coment_stmts.c
+ *
+ *	  All routines to deparse comment statements.
+ *
+ * Copyright (c), Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "catalog/namespace.h"
+#include "commands/defrem.h"
+#include "lib/stringinfo.h"
+#include "nodes/parsenodes.h"
+#include "parser/parse_type.h"
+#include "utils/builtins.h"
+#include "utils/elog.h"
+
+#include "pg_version_compat.h"
+
+#include "distributed/citus_ruleutils.h"
+#include "distributed/commands.h"
+#include "distributed/deparser.h"
+#include "distributed/listutils.h"
+#include "distributed/log_utils.h"
+
+
+typedef struct
+{
+	char *name;
+	int type;
+} ObjectTypeInfo;
+
+const ObjectTypeInfo ObjectTypeNames[] =
+{
+	[OBJECT_DATABASE] = { "DATABASE", T_String },
+	[OBJECT_ROLE] = { "ROLE", T_String },
+	[OBJECT_TSCONFIGURATION] = { "TEXT SEARCH CONFIGURATION", T_List },
+	[OBJECT_TSDICTIONARY] = { "TEXT SEARCH DICTIONARY", T_List },
+
+	/* etc. */
+};
+
+char *
+DeparseCommentStmt(Node *node)
+{
+	CommentStmt *stmt = castNode(CommentStmt, node);
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+	const char *objectName = NULL;
+	if (ObjectTypeNames[stmt->objtype].type == T_String)
```

Comment:
Instead of storing the expected type for `stmt->object` in the array, we can simply check the type of `stmt->object` using the `IsA` macro:
```suggestion
	if (IsA(stmt->object, String))
```

Reply:
Fixed